src/System.Net.Requests/src/System/Net/HttpWebRequest.cs

```diff
@@ -272,7 +274,48 @@ public int ContinueTimeout
                 _continueTimeout = value;
             }
         }
-      
+
+        public override int Timeout
+        {
+            get
+            {
+                return _timeout;
+            }
+            set
+            {
+                if (value < 0 && value != System.Threading.Timeout.Infinite)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(value), SR.net_io_timeout_use_ge_zero);
+                }
+                if (_timeout != value)
+                {
+                    _timeout = value;
+                }
```

Comment:
Nit: since nothing else is being guarded behind the if, the if can be removed and this can just set the field